catt/cli.py

```diff
@@ -103,28 +107,111 @@ def write_config(settings):
         raise CattCliError("No device specified.")
 
 
+def hunt_subtitle(video):
+    """"Searches for subtitles in the current folder"""
+    video_path = Path(video)
+    video_path_stem_lower = video_path.stem.lower()
+    for entry in video_path.parent.iterdir():
+        entry_path = Path(entry)
+        if entry_path.is_dir():
+            continue
+        if entry_path.stem.lower().startswith(video_path_stem_lower) and \
+                entry_path.suffix.lower() in [".vtt", ".srt"]:
+            return str(entry_path.resolve())
+    return None
+
+
+def convert_srt_to_webvtt_helper(content):
+    content = re.sub(r'([\d]+)\,([\d]+)', r'\1.\2', content)
+
+    with tempfile.NamedTemporaryFile(mode='w+b',
+                                     suffix=".vtt",
+                                     delete=False) as vttfile:
+        target_filename = vttfile.name
+        vttfile.write("WEBVTT\n\n".encode())
+        vttfile.write(content.encode())
+        return target_filename
+
+
+def convert_srt_to_webvtt(filename):
+    # print("Converting {} to WebVTT".format(filename))
+    for possible_encoding in ['utf-8', 'iso-8859-15']:
+        try:
+            with open(filename, 'r', encoding=possible_encoding) as srtfile:
+                content = srtfile.read()
+                return convert_srt_to_webvtt_helper(content)
+        except UnicodeDecodeError:
+            pass
+    raise CattCliError("Could not find the proper encoding of {}. Please convert it to utf-8".format(filename))
+
+
+def load_subtitle_if_exists(subtitle, video, local_ip, port):
+    subtitle = subtitle or hunt_subtitle(video)
+    if subtitle is None:
+        return None
+    click.echo("Using subtitle {}".format(subtitle))
+
+    if "://" in subtitle:
+        # it's an URL
+        if subtitle.lower().endswith(".srt"):
+            content = requests.get(subtitle).text
+            subtitle = convert_srt_to_webvtt_helper(content)
+        else:
+            return subtitle
+
+    if subtitle.lower().endswith(".srt"):
+        subtitle = convert_srt_to_webvtt(subtitle)
+
+    thr = Thread(target=serve_file,
+                 args=(subtitle, local_ip, port, "text/vtt;charset=utf-8"))
+    thr.setDaemon(True)
+    thr.start()
+    subtitle_url = "http://{}:{}/{}".format(local_ip, port, subtitle)
+    return subtitle_url
+
+
+def process_subtitle(ctx, param, value):
+    if value is None:
+        return None
+
+    if "://" in value:
+        return value
+
+    if not Path(value).is_file():
+        raise CattCliError("Subtitle file [{}] does not exist".format(value))
+
+    return value
+
+
 @cli.command(short_help="Send a video to a Chromecast for playing.")
 @click.argument("video_url", callback=process_url)
+@click.argument("subtitle", required=False, callback=process_subtitle)
```

Comment:
I think this would be better served as an option, rather than as an argument.

Reply:
I agree, but that would also change the applicability of ```--dont-load-subtitles-automatically```. Because if this was an option, you might as well just say that ```--subtitle``` without any argument means "try to automatically load subtitle".

Reply:
Yes, exactly. And `--no-subtitle` would mean "don't load, even automatically".

Reply:
The problem is that there are three states, file/auto/none. So we need `--subtitle`/nothing/`--no-subtitle`.

Reply:
Not specifying ```--subtitle``` means none.
Specifying ```--subtitle``` without an argument means auto.
Specifying ```--subtitle <subtitle>``` means file.
What am I missing?

Reply:
So:
Nothing specified means "auto".
Specifying ```--subtitle <subtitle>``` means "file".
Specifying ```--no-subtitle``` means "none".

Having given this a bit of thought, I can see that this is probably the behaviour most users would expect.

Reply:
good, we all agreed on the same thing. I just got somebody visiting me, so I'll continue later.